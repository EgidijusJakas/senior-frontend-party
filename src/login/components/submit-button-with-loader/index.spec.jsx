import React from 'react';
import { shallow } from 'enzyme';
import { SubmitButtonWithLoader } from '.';

describe('Submit Button', () => {
    it('should render button without loader', () => {
        const props = {
            isAuthenticating: false
        };

        const renderedComponent = shallow(<SubmitButtonWithLoader {...props} />);
        expect(renderedComponent).toMatchSnapshot();
    });

    it('should render button with loader', () => {
        const props = {
            isAuthenticating: true
        };

        const renderedComponent = shallow(<SubmitButtonWithLoader {...props} />);
        expect(renderedComponent).toMatchSnapshot();
    });
});
