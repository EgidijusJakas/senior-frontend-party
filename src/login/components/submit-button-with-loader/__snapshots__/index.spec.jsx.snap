// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit Button should render button with loader 1`] = `
<styled.div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <Component
    color="#fff"
    component="div"
    loaded={false}
    loadedClassName="loadedContent"
    parentClassName="loader"
  />
  <button
    className="btn btn-sm btn-block"
    type="submit"
  />
</styled.div>
`;

exports[`Submit Button should render button without loader 1`] = `
<styled.div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <Component
    color="#fff"
    component="div"
    loaded={true}
    loadedClassName="loadedContent"
    parentClassName="loader"
  />
  <button
    className="btn btn-sm btn-block"
    type="submit"
  >
    Log In
  </button>
</styled.div>
`;
